JobIntentService throws IllegalArgumentException on completeWork

**App version:** 1.6.6 Beta

**Android version**: 8.0, 8.1, 9.0

**Device model**: OnePlus, Pixel

```
java.lang.RuntimeException: 
  at android.os.AsyncTask$3.done (AsyncTask.java:353)
  at java.util.concurrent.FutureTask.finishCompletion (FutureTask.java:383)
  at java.util.concurrent.FutureTask.setException (FutureTask.java:252)
  at java.util.concurrent.FutureTask.run (FutureTask.java:271)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1162)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:636)
  at java.lang.Thread.run (Thread.java:764)
Caused by: java.lang.IllegalArgumentException: 
  at android.app.job.JobParameters.completeWork (JobParameters.java:221)
  at android.support.v4.app.JobIntentService$JobServiceEngineImpl$WrapperWorkItem.complete (JobIntentService.java:267)
  at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:393)
  at android.support.v4.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:382)
  at android.os.AsyncTask$2.call (AsyncTask.java:333)
  at java.util.concurrent.FutureTask.run (FutureTask.java:266)
```

--- a/core/src/main/java/de/danoeh/antennapod/core/service/GpodnetSyncService.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/service/GpodnetSyncService.java
@@ -45,6 +45,7 @@ import de.danoeh.antennapod.core.util.gui.NotificationUtils;
  * This class also provides static methods for starting the GpodnetSyncService.
  */
 public class GpodnetSyncService extends JobIntentService {
+
     private static final String TAG = "GpodnetSyncService";
 
     private static final long WAIT_INTERVAL = 5000L;
@@ -61,8 +62,10 @@ public class GpodnetSyncService extends JobIntentService {
     private static boolean syncSubscriptions = false;
     private static boolean syncActions = false;
 
+    private static final int JOB_ID = -17000;
+
     private static void enqueueWork(Context context, Intent intent) {
-        enqueueWork(context, GpodnetSyncService.class, 0, intent);
+        enqueueWork(context, GpodnetSyncService.class, JOB_ID, intent);
     }
 
     @Override

--- a/core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/service/PlayerWidgetJobService.java
@@ -26,13 +26,16 @@ import de.danoeh.antennapod.core.receiver.PlayerWidget;
  * Updates the state of the player widget
  */
 public class PlayerWidgetJobService extends JobIntentService {
+
     private static final String TAG = "PlayerWidgetJobService";
 
     private PlaybackService playbackService;
     private final Object waitForService = new Object();
 
+    private static final int JOB_ID = -17001;
+
     public static void updateWidget(Context context) {
-        enqueueWork(context, PlayerWidgetJobService.class, 0, new Intent(context, PlayerWidgetJobService.class));
+        enqueueWork(context, PlayerWidgetJobService.class, JOB_ID, new Intent(context, PlayerWidgetJobService.class));
     }
 
     @Override

